src/views/LoginModal.jsx

```diff
@@ -17,192 +24,228 @@ import twitterLogo from '../assets/img/twitterlogo.svg';
 import googleLogo from '../assets/img/googlelogo.svg';
 
 // Actions
-import { loginUserAction, loadingAuthAction, clearAuthErrorAction } from '../redux/actions/authActions';
+import {
+  loginUserAction,
+  loadingAuthAction,
+  clearAuthErrorAction
+} from '../redux/actions/authActions';
 
 /**
  * @description loginmodal
  * @param {object} e event object
  * @returns {HTMLDivElement} login modal
  */
 export class LoginModal extends Component {
-    state = {
-      name: '',
-      password: '',
-      rememberMe: false
-    }
+  state = {
+    name: '',
+    password: '',
+    rememberMe: false
+  };
 
-    clearErrors = () => {
-      const { clearAuthError, errorMessages } = this.props;
-      if (errorMessages.errors) {
-        clearAuthError();
-      }
+  clearErrors = () => {
+    const { clearAuthError, errorMessages } = this.props;
+    if (errorMessages.errors) {
+      clearAuthError();
     }
+  };
 
-    componentWillMount = () => {
-      this.clearErrors();
-    }
+  componentWillMount = () => {
+    this.clearErrors();
+  };
 
-    handleInputChange = (e) => {
-      this.clearErrors();
-      this.setState({
-        [e.target.name]: e.target.value
-      });
-    }
+  handleInputChange = (e) => {
+    this.clearErrors();
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
 
-    handleSubmit = (e) => {
-      e.preventDefault();
-      const {
-        loginUser, loadingAuth
-      } = this.props;
-      loadingAuth();
-      loginUser(this.state);
-    }
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { loginUser, loadingAuth } = this.props;
+    loadingAuth();
+    loginUser(this.state);
+  };
 
-    doNothing = () => ({ message: 'Processing Request' })
+  doNothing = () => ({ message: 'Processing Request' });
 
-    handleErrorMessages = () => {
-      const {
-        errorMessages
-      } = this.props;
-      return errorMessages.errors ? {
+  handleErrorMessages = () => {
+    const { errorMessages } = this.props;
+    return errorMessages.errors
+      ? {
         nameError: errorMessages.errors.name ? errorMessages.errors.name[0] : '',
         passwordError: errorMessages.errors.password ? errorMessages.errors.password[0] : '',
-        generalError: errorMessages.errors.general || '',
-      } : {
+        generalError: errorMessages.errors.general || ''
+      }
+      : {
         nameError: '',
         passwordError: '',
         generalError: ''
       };
-    };
+  };
+
+  /**
+   * @description social Login
+   * @param {string} URL
+   * @returns {null} null
+   */
+  socialMediaLogin = (URL) => {
+    if (URL === GOOGLE_SOCIAL_LOGIN_URL) {
+      window.location.assign(GOOGLE_SOCIAL_LOGIN_URL);
+    } else if (URL === FACEBOOK_SOCIAL_LOGIN_URL) {
+      window.location.assign(FACEBOOK_SOCIAL_LOGIN_URL);
+    } else {
+      window.location.assign(TWITTER_SOCIAL_LOGIN_URL);
+    }
```

Comment:
Why not just use `window.location.assign(URL)`?

Reply:
It redirects the user

Reply:
I have removed the unnecessary line of code.